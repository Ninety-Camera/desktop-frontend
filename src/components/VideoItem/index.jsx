import React from "react";
import { useState } from "react";
import { Button, CircularProgress, Stack } from "@mui/material";
import { useDispatch, useSelector } from "react-redux";
import ToggleBtn from "../ToggleBtn";
import { LOCAL_URL } from "../../constants";
import { deleteCamera } from "../../reducers/cameraSlice";
import HeightBox from "../HeightBox";

export default function VideoItem(props) {
  const videoClip = props.videoClip;
  const dispatch = useDispatch();
  const userState = useSelector((state) => state.user);
  const [state, setState] = useState("RUNNING");
  const [loading, setLoading] = useState(false);

  async function delCamera() {
    setLoading(true);
    dispatch(
      deleteCamera({
        camId: videoClip?.id,
        token: userState?.token,
      })
    );
  }

  return (
    <div
      style={{
        backgroundColor: "white",
        boxShadow:
          "0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
        marginBottom: "25px",
      }}
    >
      <img
        src={`${LOCAL_URL}video_feed/${videoClip?.id}`}
        width={320}
        height={240}
        alt="CCTV video"
        onError={(event) => {
          event.target.src =
            "https://www.svgrepo.com/show/343419/computer-error.svg";
          event.onerror = null;
        }}
        style={{
          borderRadius: "8px",
          padding: "5px",
          width: "95%",
          margin: "10px",
        }}
      />

      <div>
        <Stack direction="column" spacing={2} sx={{ alignContent: "justify" }}>
          <p
            style={{
              color: "black",
              fontSize: 20,
              margin: 0,
              textAlign: "center",
              fontWeight: 800,
            }}
          >
            {videoClip?.name}
          </p>
          <div
            style={{
              justifyContent: "center",
              marginLeft: "auto",
              marginRight: "auto",
              textAlign: "center",
              padding: "10px 20px",
            }}
          >
            <Stack direction="row" spacing={1}>
              <ToggleBtn state={state} setState={setState} />
              <Button
                variant="contained"
                onClick={delCamera}
                disabled={loading}
              >
                {loading ? <CircularProgress /> : "Delete Camera"}
              </Button>
            </Stack>
          </div>
          {/* <HeightBox height={10} />
        <div
          style={{
            justifyContent: "center",
            marginLeft: "auto",
            marginRight: "auto",
          }}
        >
          
        </div> */}
        </Stack>
      </div>
    </div>
  );
}
