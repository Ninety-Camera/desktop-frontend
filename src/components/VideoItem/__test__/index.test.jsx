import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import ReactDom from "react-dom";
import VideoItem from "..";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { useState } from "react";
import VIDEOCLIP1 from "../../../assets/video1.mp4";

const videoClip = VIDEOCLIP1;

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDom.render(<VideoItem videoClip={videoClip} index={1} />, div);
});


it("matches snapshot", () => {
  const tree = renderer
    .create(<VideoItem videoClip={videoClip} index={1} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
