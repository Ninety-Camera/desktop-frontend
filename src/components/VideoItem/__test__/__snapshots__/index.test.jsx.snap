// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <img
    alt="CCTV video"
    height={240}
    onError={[Function]}
    width={320}
  />,
  <div
    className="css-aj204s-MuiStack-root"
  >
    <p
      style={
        Object {
          "color": "#6C63FF",
          "fontSize": 20,
          "fontWeight": 800,
          "margin": 0,
          "textAlign": "center",
        }
      }
    >
      CCTV - 2
    </p>
    <div
      style={
        Object {
          "justifyContent": "center",
          "marginLeft": "auto",
          "marginRight": "auto",
        }
      }
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        data-testid="toggleBtn"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "textTransform": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="StopIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 6h12v12H6z"
            />
          </svg>
        </span>
        Stop Monitoring
      </button>
    </div>
  </div>,
]
`;
